plenum/server/req_authenticator.py

```diff
@@ -35,7 +36,7 @@ def authenticate(self, req_data):
             if not (authenticator.is_write(typ) or
                     authenticator.is_action(typ)):
                 continue
-            rv = authenticator.authenticate(req_data) or set()
+            rv = authenticator.authenticate(deepcopy(req_data)) or set()
```

Comment:
Why do we need to make a copy here? Why do we modify req_data during authentication check?